cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -141,6 +145,10 @@ public synchronized void create(final NamespaceMeta metadata) throws Exception {
       authorizerInstantiator.get().enforce(instanceId, principal, Action.ADMIN);
     }
 
+    // Skip authorization grants for the system user
+    if (!(Principal.SYSTEM.equals(principal) && NamespaceId.DEFAULT.equals(namespace))) {
```

Comment:
so we skip for the case when the user is system and the namespace is default? so creation of default namespace happens during the cdap startup, isn't it? in that case, what will be the Principal? during the normal run, we don't allow creation of 'default' namespace anyway. can you expand the comment a bit more to explain what this check doing?